### DISCUSSION TITLE
Some things to ask about mobile developments with Avalonia

### DISCUSSION THREAD

**Discussion by shanoaice** [AvaloniaUI/Avalonia#6599] [Q&A]
This discussion actually started here: [AvaloniaUI/Avalonia#1602](https://github.com/AvaloniaUI/Avalonia/issues/1602#issuecomment-918328461). I think it may be inappropriate to ask questions there, so I moved the quesions here:
1. It seemed like the `Avalonia.Android` NuGet package is abandoned. Is it essential to install `Avalonia.Android` package to develop application for Android or should I just simply reference the UI project from the `Xamarin.[iOS, Android]` projects and then I'm ready to go?
2. I uses `Avalonia.FuncUI` to write applications in MVU style, so there may be a subtle difference: the application does override as follows:
```fsharp
override this.OnFrameworkInitializationCompleted() =
        match this.ApplicationLifetime with
        | :? IClassicDesktopStyleApplicationLifetime as desktopLifetime ->
            desktopLifetime.MainWindow <- Shell.MainWindow()
        | _ -> ()
```
So here I just add a case `ISingleViewApplicationLifetime ` and register as `singleViewLifetime.MainWindow` and everyting will be operational. Is that correct?

3. Also I am curious about IME support on mobile devices -- althought not very essential to my purpose though (people usually don't mark their Shadowsocks/v2ray nodes with Chinese or other IME dependent language, but who knows?). I see pull requests adding support for Win32 IME and I am hoping to see that being merged, and I also see some pull requests adding basic soft keyboard input support, but I am unsure if a input method is usable.
4. One last thing: could there be some very serious compatibility issues that I may not be able to use some component libraries (ex. Material Design, Material Design Icons, Charts)? (I understand that Avalonia Mobile is not for production use and I can tolerate bugs. After all, my application is also not a serious production-ready-grade project, just for hobby. But, I am still curious about usability of some basic functionalities)

Thanks for all developers who contributed and constructed such a complex and useful library, and thank you for taking all these time to answer my questions.

**Comment by shanoaice** [comment]
@kekekeks Sorry for this interruption, but I still have some small question to ask and I thought it is not appropriate to ask in that issue, so I put my question here.

**Comment by maxkatz6** [comment]
1. Avalonia.Android wasn't updated in a while because its uploading was disabled because of instability. Right now it works way better, but it's considered as a production ready. So you basically need to build Avalonia by yourself to use latest android package.
2. Yes, ISingleViewApplicationLifetime  is enough. But it's possible that some third-party controls might heavily depend on IClassicDesktopStyleApplicationLifetime or ```control.VisualRoot as Window``` (there is no 'window' for mobile apps, only root view). 
3. Can't answer that. Maybe @kekekeks knows. As far as I know it's possible now to implement IME for mobiles, but nobody did it yet.
4. See number 2. It's possible, but unlikely with themes/charts controls. Because they can rely on desktop-only APIs.
